app/controllers/session_controller.rb

```diff
@@ -33,8 +49,25 @@ def create
     redirect_after_sucessful_login
   end
 
+  def sso
```

Comment:
please move this to own controller as already discussed:

Session::SsoController#create